patches/net/minecraft/client/renderer/block/LiquidBlockRenderer.java.patch

```diff
@@ -1,15 +1,36 @@
 --- a/net/minecraft/client/renderer/block/LiquidBlockRenderer.java
 +++ b/net/minecraft/client/renderer/block/LiquidBlockRenderer.java
-@@ -37,6 +_,7 @@
+@@ -37,12 +_,17 @@
          this.waterIcons[0] = Minecraft.getInstance().getModelManager().getBlockModelShaper().getBlockModel(Blocks.WATER.defaultBlockState()).getParticleIcon();
          this.waterIcons[1] = ModelBakery.WATER_FLOW.sprite();
          this.waterOverlay = ModelBakery.WATER_OVERLAY.sprite();
 +        net.neoforged.neoforge.client.textures.FluidSpriteCache.reload();
      }
  
      private static boolean isNeighborSameFluid(FluidState p_203186_, FluidState p_203187_) {
-@@ -70,8 +_,9 @@
+         return p_203187_.getType().isSame(p_203186_.getType());
+     }
+ 
++    private static boolean isNeighborStateHidingOverlay(FluidState selfState, BlockState otherState) {
++        return otherState.shouldHideAdjacentFluidFace(selfState);
++    }
++
+     private static boolean isFaceOccludedByState(Direction p_110980_, float p_110981_, BlockState p_110983_) {
+         VoxelShape voxelshape = p_110983_.getFaceOcclusionShape(p_110980_.getOpposite());
+         if (voxelshape == Shapes.empty()) {
+@@ -64,14 +_,20 @@
+         return isFaceOccludedByState(p_110963_.getOpposite(), 1.0F, p_110962_);
+     }
  
++    @Deprecated // Neo: use overload that accepts BlockState
+     public static boolean shouldRenderFace(FluidState p_203169_, BlockState p_203170_, Direction p_203171_, FluidState p_203172_) {
+         return !isFaceOccludedBySelf(p_203170_, p_203171_) && !isNeighborSameFluid(p_203169_, p_203172_);
+     }
+ 
++    public static boolean shouldRenderFace(FluidState p_203169_, BlockState p_203170_, Direction p_203171_, BlockState otherState) {
```

Comment:
This method should use proper parameter names since it doesn't need to avoid changing otherwise unmodified vanilla code.

Reply:
That's a fair point, I completely forgot that JST is a bit less strict when it looks up parameter names to apply to overload methods, though I'm not totally sure this case is actually considered an overload by JST.